delta-api/src/main/java/io/cdap/delta/api/EventReader.java

```diff
@@ -32,6 +32,21 @@
    * @throws InterruptedException if the reader was interrupted while stopping
    */
   default void stop() throws InterruptedException {
+    stop(StopReason.UNKNOWN);
+  }
+
+  /**
+   * Stop reading events.
+   * @param reason the reason why stop is requested.
+   * @throws InterruptedException if the reader was interrupted while stopping
+   */
+  default void stop(StopReason reason) throws InterruptedException {
```

Comment:
Would be good to have wrapper class `StopContext`  so that future additions can be done.
```
interface StopContext {
  
   public enum Origin {
      UNKNOWN,
      ERROR, // stop initiated because error
      USER // stop initiated by user
   }

   Origin getOrigin();
}
```

Then stop method could be 
```
default void stop(StopContext context) throws InterruptedException {}...
```